lib/input_parser.ex

```diff
@@ -0,0 +1,78 @@
+defmodule InputParser do
+  @exit_command "exit"
+  @flag_command "-f"
+
+  def exit_command, do: @exit_command
+
+  def parse_turn(input) do
+    find_turn_input_type(input)
+    |> parse()
+  end
+
+  def parse_number(input) do
+    cond do
+      is_exit_command?(input) ->
+        {:exit, input}
+
+      true ->
+        {:number, input}
+    end
+  end
+
+  defp find_turn_input_type(input) do
+    cond do
+      is_exit_command?(input) ->
+        {:exit, input}
+
+      has_flag?(input) && has_number_and_letter?(input) ->
+        {:flag, input}
+
+      has_number_and_letter?(input) ->
+        {:move, input}
+
+      true ->
+        {:error, :missing_letter_and_number}
```

Comment:
I still have to make use of this error in `Game`. That will also be a part of my "validations" PR